### PR TITLE
Crypto: Implementation of Diffie-Hellman key exchange

### PR DESCRIPTION
**Pull request by ec-m**
The checks give me as a failed example:

> prk = dh_private_key(); prk
> Expected:
> (1031, 14, 66)
> Got:
> (1031, 14, 176)

which is probably due to the fact that the last element is a random number in between 2 and 1030. How can I avoid the checks failing if this is what is suppose to happen?

### CODE REVIEW

**Comment by moorepants** [comment]
Is there a random number generator seed you can set?

**Comment by ec-m** [comment]
@moorepants  I'll try that. Thanks!

File: sympy/crypto/crypto.py [hunk 1]
```diff
@@ -1589,3 +1589,126 @@ def decipher_elgamal(ct, prk):
...
+    """
+    p, g, a = prk
+    return p, g, pow(g, a, p)
+
+def dh_shared_key(puk, b):
+    """
+    Return int as shared key.
+
+    This is what Bob and Alice can both calculate using the public
+    keys their received from each other and their private keys.
```
Comment (smichr) [inline sympy/crypto/crypto.py]: their received -> they received

**Comment by smichr** [comment]
> random number generator seed you can set

`_randint` was added for this purpose. See how, for example, it is used in `combinatorics/partitions.py`

File: sympy/crypto/crypto.py (hunk 1 above)
Comment (ec-m) [inline sympy/crypto/crypto.py]: I changed that. Thank you!

**Comment by ec-m** [comment]
The build issues laid within the examples of the docstring. I now fixed these problems without using a random number generator seed by simply adjusting the examples. @moorepants @smichr

**Comment by certik** [comment]
That looks good to me, thanks!

**Pull request merged**